File: Data Structures & Algorithms/find-peak-element/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findPeakElement(nums) {

        if(nums.length === 1) {
            return 0;
        }

        if(nums.length > 1 && (nums[nums.length - 1] > nums[nums.length - 2])) {
            return nums.length - 1;
        }

        if(nums.length > 1 && (nums[0] > nums[1])) {
            return 0;
        }


        for(let i = 1; i < nums.length - 1; i += 1) {
            if(nums[i] > nums[i - 1] && nums[i] > nums[i + 1]) {
                return i;
            }
        }
    }
}

File: Data Structures & Algorithms/remove-element/submission-4.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        
        i = 0

        while i < len(nums):
            if nums[i] == val:
                j = i + 1
                while j < len(nums) and nums[j] == val:
                    j += 1
                if j >= len(nums):
                    return i
                nums[i] = nums[j]
                nums[j] = val
            i += 1
        
        return i
